src/Database/Validator/Permissions.php

```diff
@@ -32,10 +33,16 @@ class Permissions extends Validator
      * Permissions constructor.
      *
      * @param int $length maximum amount of permissions. 0 means unlimited.
+     * @param array $allowedPermissions
      */
-    public function __construct(int $length = 0)
+    public function __construct(int $length = 0, array $allowedPermissions = [])
```

Comment:
```suggestion
    public function __construct(int $length = 0, array $allowed = [])
```

We're already under the permissions namespace. $allowed should be good enough

Reply:
Same for the protected property.